### PR TITLE
Revamp text editor for homepage

### PR DESCRIPTION
**Pull request by vgoh1**
# Description
One of the users we talked to suggested having a text editor with formatting options for the homepage. This was implemented using [react-draft-wysiwyg](https://github.com/jpuri/react-draft-wysiwyg).

# Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Refactoring
- [ ] Documentation
- [x] Improvement

# Solution description
The homepage model has been changed to store a (HTML) string instead of an array of strings. Integration tests have been updated accordingly.
The about us text is stored as an HTML string.
Scholly/base needs a HTML parser to be able to take the string and display it on the React side.

# Tests

Feel free to test as usual as I don't think I handled all of the edge cases. **FYI: the app will throw an error if your homepage doc aboutUs field is an array, so you should change it to a string before testing. At the time of writing this, I already changed it in the db.**

image

When deployed:
image

# Relevant document/ link (if any)
Draftjs toolbar options: https://jpuri.github.io/react-draft-wysiwyg/#/docs?_k=jjqinp

# Risk

Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
Low, affects homepage editor only

### CODE REVIEW

File: api/src/models/homepage.model.js [hunk 1]
```diff
@@ -11,13 +11,11 @@ const homepageSchema = new mongoose.Schema(
...
     },
-    // `aboutUs` stores a list of strings.
-    // Each string in list is a paragraph.
-    aboutUs: [
-      {
-        type: String,
-      },
-    ],
+    // `aboutUs` stores a HTML string
+    aboutUs:
```
Comment (aksh0001) [inline api/src/models/homepage.model.js:15]: Could you update the trigger in the team model as well? I believe we’re defaulting to an empty array, which should now be an empty string. You can replace the empty array assertion with empty string in the test as well

File: client/src/reducers/homepageReducer.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { FETCH_HOMEPAGE, UPDATE_HOMEPAGE } from '../actions/types';
 // Each string in list is representing a paragraph.
 // This is for better handling the display/rendering of each Para.
 const INITIAL_HOMEPAGE_STATE = {
-  aboutUs: [''],
+  aboutUs: '',
```
Comment (aksh0001) [inline client/src/reducers/homepageReducer.js:10]: make sure you update the doc above to reflect the new data structure

File: scholly/base/src/layouts/layout1/components/landingPage/LandingPage.js [hunk 3]
```diff
@@ -4,6 +4,7 @@
 import React from 'react';
 import { Container } from 'react-bootstrap';
 import { Helmet } from 'react-helmet';
+import parse from 'html-react-parser';
```
Comment (aksh0001) [inline scholly/base/src/layouts/layout1/components/landingPage/LandingPage.js:7]: nit: `\n` after this

File: scholly/base/src/layouts/layout2/components/landingPage/LandingPage.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ import React from 'react';
 import { Col, Image, Row } from 'react-bootstrap';
 import { Helmet } from 'react-helmet';
 import { Link } from 'react-router-dom';
+import parse from 'html-react-parser';
```
Comment (aksh0001) [inline scholly/base/src/layouts/layout2/components/landingPage/LandingPage.js:8]: `\n`

File: scholly/base/src/layouts/layout3/components/landingPage/LandingPage.js [hunk 5]
```diff
@@ -4,6 +4,7 @@
 import React from 'react';
 import { Container } from 'react-bootstrap';
 import { Helmet } from 'react-helmet';
+import parse from 'html-react-parser';
```
Comment (aksh0001) [inline scholly/base/src/layouts/layout3/components/landingPage/LandingPage.js:7]: ditto

**Review by aksh0001 — APPROVED** [review]
